pkg/registry/apis/folders/folder_storage.go

```diff
@@ -157,5 +158,10 @@ func (s *folderStorage) setDefaultFolderPermissions(ctx context.Context, orgID i
 		}...)
 	}
 	_, err := s.folderPermissionsSvc.SetPermissions(ctx, orgID, uid, permissions...)
-	return err
+	if err != nil {
+		return err
+	}
+
+	s.acService.ClearUserPermissionCache(user)
```

Comment:
I think we can limit this to users:
```suggestion
	if user.IsIdentityType(claims.TypeUser) {
		s.acService.ClearUserPermissionCache(user)
	}
```